File: DropDownList/MultiSelectDropDown/MultiSelectDropDownVB/CustomListDataItem.vb
Imports Telerik.WinControls.UI
Imports Telerik.WinControls
Imports System.ComponentModel

Public Class CustomListDataItem
    Inherits RadListDataItem
#Region "RadProperties"

    Public Shared ReadOnly CheckedProperty As RadProperty = RadProperty.Register("Checked", GetType(Boolean), GetType(CustomListDataItem), New RadElementPropertyMetadata(False))

#End Region

#Region "Properties"

    Public Property Checked() As Boolean
        Get
            Return CBool(Me.GetValue(CustomListDataItem.CheckedProperty))
        End Get
        Set(value As Boolean)
            Me.SetValue(CustomListDataItem.CheckedProperty, value)
        End Set
    End Property

    Protected Overrides Sub OnPropertyChanged(e As RadPropertyChangedEventArgs)
        MyBase.OnPropertyChanged(e)
    End Sub

#End Region

#Region "Overrides"

    Protected Overrides Sub SetDataBoundItem(dataBinding As Boolean, value As Object)
        MyBase.SetDataBoundItem(dataBinding, value)
        If TypeOf value Is INotifyPropertyChanged Then
            Dim item As INotifyPropertyChanged = TryCast(value, INotifyPropertyChanged)
            AddHandler item.PropertyChanged, AddressOf item_PropertyChanged
        End If
    End Sub

#End Region

#Region "Private Methods"

    Private Sub item_PropertyChanged(sender As Object, e As PropertyChangedEventArgs)
        If e.PropertyName = "Checked" Then
            Me.Checked = TryCast(Me.DataBoundItem, RadListDataItem).Selected
        End If
    End Sub
#End Region
End Class

File: GridView/Multiselect-Column-cs/Multiselect-Column-cs/CustomListDataItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Telerik.WinControls.UI;
using Telerik.WinControls;
using System.ComponentModel;

namespace _547099
{
    public class CustomListDataItem : RadListDataItem
    {
        #region RadProperties

        public static readonly RadProperty CheckedProperty = RadProperty.Register("Checked", typeof(bool), typeof(CustomListDataItem), new RadElementPropertyMetadata(false));
        
        #endregion

        #region Properties

        public bool Checked
        {
            get
            {
                return (bool)this.GetValue(CustomListDataItem.CheckedProperty);
            }
            set
            {
                this.SetValue(CustomListDataItem.CheckedProperty, value);
            }
        }

       

        #endregion

        #region Overrides

        protected override void SetDataBoundItem(bool dataBinding, object value)
        {
            base.SetDataBoundItem(dataBinding, value);
            if (value is INotifyPropertyChanged)
            {
                INotifyPropertyChanged item = value as INotifyPropertyChanged;
                item.PropertyChanged += item_PropertyChanged;
            }
        }

        #endregion

        #region Private Methods

        private void item_PropertyChanged(object sender, PropertyChangedEventArgs e)
        {
            if (e.PropertyName == "Checked")
            {
                this.Checked = (this.DataBoundItem  as RadListDataItem).Selected;
            }            
        }

        #endregion
    }
}
